arbor/fvm_layout.cpp

```diff
@@ -561,20 +565,95 @@ fvm_mechanism_data fvm_build_mechanism_data(const mechanism_catalogue& catalogue
 
         fvm_mechanism_config& config = mechdata.mechanisms[name];
         config.kind = mechanismKind::point;
+        config.linear = catalogue[name].linear;
+
+        // Generate config.cv_loc: maps synapse instance to cv location in config.cv
+        // Generate config.cv: contains cv of group of synapses that can be coalesced into one instance
+        // Generate config.param_values: contains parameters of group of synapses that can be coalesced into one instance
+        // Generate coalesced_mult: contains number of synapses in each coalesced group of synapses
+        // Generate target: contains the synapse target number
+        if(config.linear && coalesce_syanpses) {
+
+            // cv_param_vec used to lexicographically sort the cv, parameters and target, which are stored in that order
+            std::vector<std::vector<value_type>> cv_param_vec(cv_order.size(), std::vector<value_type>(nparam + 2));
+            for (unsigned i = 0; i < cv_order.size(); ++i) {
+                auto loc = cv_order[i];
+                cv_param_vec[i][0] = points[loc].cv;
+                for (auto pidx: make_span(0, nparam)) {
+                    value_type pdefault = param_default[pidx];
+                    const std::string& pname = param_name[pidx];
+                    cv_param_vec[i][pidx+1] = value_by_key(points[loc].desc->values(), pname).value_or(pdefault);
+                }
+                cv_param_vec[i][nparam + 1] = points[loc].target_index;
+            }
 
-        assign(config.cv,     transform_view(cv_order, [&](size_type j) { return points[j].cv; }));
-        assign(config.target, transform_view(cv_order, [&](size_type j) { return points[j].target_index; }));
+            std::sort(cv_param_vec.begin(), cv_param_vec.end());
 
-        config.param_values.resize(nparam);
-        for (auto pidx: make_span(0, nparam)) {
-            value_type pdefault = param_default[pidx];
-            const std::string& pname = param_name[pidx];
+            auto identical_synapse = [&nparam, &cv_param_vec](unsigned i, unsigned j){
+                bool ret = true;
+                for(unsigned idx = 0; idx < nparam + 1; idx++) {
+                    ret &= cv_param_vec[i][idx] == cv_param_vec[j][idx];
+                    if (!ret) break;
+                }
+                return ret;
+            };
+
+            // Initialize vars
+            unsigned loc = 0;
+            unsigned count = 1;
+            bool coalesce = true;
+
+            config.param_values.resize(nparam);
+            config.cv_loc.reserve(cv_param_vec.size());
+            config.target.reserve(cv_param_vec.size());
+
+            config.cv_loc.push_back(loc);
+            config.target.push_back((index_type)cv_param_vec.front()[1 + nparam]);
+
+            for (unsigned i = 1; i < cv_param_vec.size(); ++i) {
```

Comment:
Because this is iterating over a `std::vector`, it's better to use `std::size_t` for consistency.
That said, this might be simplified some by the use of iterators and `util::range` e.g.
```
for (auto i = cv_param_vec.begin(), j = i; i!=cv_param_vec.end(); i = j) {
    ++j;
    while (j!=cv_param_vec.end() && identical(*i, *j)) ++j;

    auto mergeable = util::make_range(i, j);
    // mergeable is a range view corresponding to the identical synapses.
    // ...
}
```
where `identical` is the corresponding equality test.

Reply:
Done. Used suggested code.